client/lib/with-dimensions/index.jsx

```diff
@@ -5,30 +5,32 @@ import React from 'react';
 import { debounce } from 'lodash';
 
 /**
- * withWidth is a Hoc that hands down a width prop of how much available width there is for it to consume.
- * withWidth assumes that you care about the space at the dom location of the component, but you may also pass
- * in a domTarget in case you want to be tracking the width of a different component
+ * withDimensions is a Hoc that hands down a width and height prop of how much available space there is for it to consume.
+ * withDimensions assumes that you care about the space at the dom location of the component, but you may also pass
+ * in a domTarget in case you want to be tracking the dimensions of a different component
  *
  * @example:
- * 1. widthWidth( Component )
- * 2. withWidth( Component, { domTarget: thingWhoseWidthToTrack } )
+ * 1. withDimensions( Component )
+ * 2. withDimensions( Component, { domTarget: thingWhoseDimensionsToTrack } )
  *
- * @param {object} EnhancedComponent - react component to wrap and give the prop width to
+ * @param {object} EnhancedComponent - react component to wrap and give the prop width/height to
  * @returns {object} the enhanced component
  */
 export default ( EnhancedComponent, { domTarget } = {} ) => class WithWidth extends React.Component {
-	static displayName = `WithWidth( ${ EnhancedComponent.displayName } )`;
+	static displayName = `WithDimensions( ${ EnhancedComponent.displayName || EnhancedComponent.name } )`;
 
 	state = {
 		width: 0,
+		height: 0,
 	};
 
 	handleResize = () => {
 		const domElement = domTarget ? domTarget : this.divRef;
 
 		if ( domElement ) {
-			const width = domElement.getClientRects()[ 0 ].width;
-			this.setState( { width } );
+			const dimensions = domElement.getClientRects()[ 0 ];
+			const { width, height } = dimensions;
+			this.setState( { width, height } );
```

Comment:
Thanks for noticing this! It unearthed less-than-optimal api i had written.  Instead of taking `domTarget` as a param for the HoC function, it should be a prop for the generated component.  That way we aren't inventing a new component on every render